spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/PublisherCallbackChannelImpl.java

```diff
@@ -691,32 +687,33 @@ private synchronized void generateNacksForPendingAcks(String cause) {
 				logger.debug("Added listener " + listener);
```

Comment:
Oops, I should not have removed the `synchronized` here. But, yes, I can use putIfAbsent() instead - much better.